src/org/python/core/PyByteArray.java

```diff
@@ -1343,6 +1324,8 @@ final synchronized PyObject bytearray___iadd__(PyObject o) {
         } else if (oType == PyString.TYPE) {
             // Will fail if somehow not 8-bit clean
             setslice(size, size, 1, (PyString)o);
+        } else if (o instanceof BufferProtocol && !(o instanceof PyUnicode)) {
```

Comment:
Suggest `if (setsliceFromBuffer(...)) { /* we're done "/ }. I notice I haven't documented the return on that properly.

Reply:
Ah yes I didn't spot that. Will add the doc too.